Makefile

```diff
@@ -1,88 +1,112 @@
+.DEFAULT_GOAL := help
+.PHONY: help start stop reset db new-migration watch clear clean test tu tf tj lint ls ly lt lj build up perm
 
-all: install
-
-help:           ## Show this help
+help:
 	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
 
-install:        ## [start deps db-create, db-fixtures] Setup the project using Docker and docker-compose
-install: start deps db-reset perm
+start:          ## Install and start the project
+start: build up db web/built perm
+
+stop:           ## Remove docker containers
+	$(FIG) kill
+	$(FIG) rm -v --force
+
+reset:          ## Reset the whole project
+reset: stop start
+
+db:             ## Reset the database and load fixtures
+db: vendor
+	$(RUN) php -r "for(;;){if(@fsockopen('db',3306)){break;}}" # Wait for MySQL
+	$(RUN) $(CONSOLE) doctrine:database:drop --force --if-exists
+	$(RUN) $(CONSOLE) doctrine:database:create --if-not-exists
+	$(RUN) $(CONSOLE) doctrine:migrations:migrate -n
+	$(RUN) $(CONSOLE) doctrine:fixtures:load -n
+
+:##---------------------------------------------------------------------------:
+
+new-migration:  ## Generate a migration by comparing your current database to your mapping information.
+new-migration: vendor
+	$(RUN) $(CONSOLE) doctrine:migration:diff
+
+watch:          ## Watch the assets and build their development version on change
+watch: node_modules
+	$(RUN) yarn watch
 
-start:          ## Start the Docker containers
-	docker-compose up -d
+clear:          ## Remove all the cache, the logs, the sessions and the built assets
+clear: perm
+	-$(EXEC) rm -rf var/cache/*
+	-$(EXEC) rm -rf var/sessions/*
+	-$(EXEC) rm -rf supervisord.log supervisord.pid npm-debug.log .tmp
+	rm -rf var/logs/*
+	rm -rf web/built
 
-stop:           ## Stop the Docker containers
-	docker-compose down
+clean:          ## Clear and remove dependencies
+clean: clear
+	rm -rf vendor node_modules
 
-deps:           ## [composer yarn assets-dev perm] Install the project PHP and JS dependencies
-deps: composer yarn assets-dev perm
+:##---------------------------------------------------------------------------:
 
-composer:       ## Install the project PHP dependencies
-	$(TOOLS) composer install
+test:           ## Run the PHP and the Javascript tests
+test: tu tf tj
 
-yarn:           ## Install the project JS dependencies
-	$(TOOLS) yarn install
+tu:             ## Run the PHP unit tests
+tu: vendor
+	$(RUN) vendor/bin/phpunit --exclude-group functionnal
 
-db-create:      ## Create the database
-	$(TOOLS) php -r "for(;;){if(@fsockopen('db',3306)){break;}}" # Wait for MariaDB
-	$(CONSOLE) doctrine:database:drop --force --if-exists
-	$(CONSOLE) doctrine:database:create --if-not-exists
-	$(CONSOLE) doctrine:migrations:migrate -n
+tf:             ## Run the PHP functional tests
+tf: vendor
+	$(RUN) vendor/bin/phpunit --group functionnal
```

Comment:
the functional group is now called `functional`. I fixed the typo in all tests already.